src/tcgwars/logic/impl/gen8/BattleStyles.groovy

```diff
@@ -2553,7 +2551,16 @@ public enum BattleStyles implements LogicCardInfo {
         bwAbility "Stance Change", {
           text "Once during your turn (before your attack), you may switch this Pokémon with an Aegislash in your hand. (Any cards attached to this Pokémon, damage counters, Special Conditions, turns in play, and any other effects remain on the new Pokémon.)"
           actionA {
-            // TODO
+            checkLastTurn()
+            assert hand.filterByNameEquals("Aegislash")
+            powerUsed()
+            def card = hand.filterByNameEquals("Aegislash").select("Stance Change").first()
+            def old = self.topPokemonCard
+            self.cards.remove(old)
+            hand.remove(card)
+            self.cards.add(card)
+            hand.add(old)
```

Comment:
This will not be handled properly by Scoop Up Block. Current Ageislash should be discarded instead of put to hand if Scoop Up Block is active, but the one in the hand should still replace it.

Reply:
I think the new code I've added will address this :)

Reply:
Now you're just discarding the top card instead of putting it in your hand from what I see. 

You should either try to `moveTo` the hand and check if the card is still in the `self.cards` list and discard it if it is, or try to `moveCard` and it should return `true` on failure and `false` on success and you can discard the card if it fails I think.